File: Android Studio/JugnooNewUI/jugnooNewUI/src/main/java/product/clicklabs/jugnoo/home/adapters/ScheduleRideVehicleListAdapter.java
package product.clicklabs.jugnoo.home.adapters;

import android.content.Context;
import android.content.Intent;
import android.support.constraint.ConstraintLayout;
import android.support.v7.widget.RecyclerView;
import android.text.Html;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageButton;
import android.widget.ImageView;
import android.widget.RelativeLayout;
import android.widget.TextView;

import com.squareup.picasso.Picasso;

import java.util.List;

import product.clicklabs.jugnoo.Data;
import product.clicklabs.jugnoo.R;
import product.clicklabs.jugnoo.home.HomeActivity;
import product.clicklabs.jugnoo.home.models.Region;
import product.clicklabs.jugnoo.utils.Utils;

/**
 * Created by mohitkr.dhiman on 12/09/18.
 */

public class ScheduleRideVehicleListAdapter extends RecyclerView.Adapter<ScheduleRideVehicleListAdapter.ViewHolder> {
    private HomeActivity activity;
    private List<Region> vehicleList;

    public ScheduleRideVehicleListAdapter(HomeActivity activity, List<Region> vehicleList) {
        this.activity = activity;
        this.vehicleList = vehicleList;
    }

    @Override
    public ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View taskItem = ((LayoutInflater) activity.getSystemService(Context.LAYOUT_INFLATER_SERVICE)).inflate(R.layout.list_item_schedule_ride_vehicles, parent, false);
        return new ViewHolder(taskItem);
    }

    @Override
    public void onBindViewHolder(final ViewHolder holder, final int position) {
        final int adapterPos = holder.getAdapterPosition();
        for (int i = 0; i < vehicleList.size(); i++) {
            if(activity.selectedIdForScheduleRide==vehicleList.get(adapterPos).getRegionId()) {
                holder.ivSelected.setVisibility(View.VISIBLE);
            }
            else {
                holder.ivSelected.setVisibility(View.GONE);
            }
        }
        holder.tvVehicleName.setText(vehicleList.get(adapterPos).getRegionName());
        holder.tvBaseFare.setText(activity.getString(R.string.base_fare_format," "+Utils.formatCurrencyValue(vehicleList.get(adapterPos).getFareStructure().getCurrency(),vehicleList.get(adapterPos).getFareStructure().getDisplayBaseFare(activity))));
        holder.tvFarePerMinute.setText("Per Min: "+Utils.formatCurrencyValue(vehicleList.get(adapterPos).getFareStructure().getCurrency(), vehicleList.get(adapterPos).getFareStructure().farePerMin, false));
        holder.tvFarePerMile.setText(activity.getString(R.string.per_format, Utils.getDistanceUnit(vehicleList.get(adapterPos).getFareStructure().getDistanceUnit()))+": "+Utils.formatCurrencyValue(vehicleList.get(adapterPos).getFareStructure().getCurrency(), vehicleList.get(adapterPos).getFareStructure().farePerKm, false));

        Picasso.with(activity)
                .load(vehicleList.get(adapterPos).getImages().getRideNowNormal())
                .into(holder.ivVehicleImage);
        holder.clRoot.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                activity.selectedIdForScheduleRide=vehicleList.get(adapterPos).getRegionId();
                activity.selectedRideTypeForScheduleRide=vehicleList.get(adapterPos).getRideType();
                activity.selectedRegionForScheduleRide=vehicleList.get(adapterPos);
                notifyDataSetChanged();
            }
        });
    }

    @Override
    public int getItemCount() {
        return vehicleList.size();
    }

    class ViewHolder extends RecyclerView.ViewHolder {


        TextView tvVehicleName, tvBaseFare, tvFarePerMinute, tvFarePerMile;
        ImageView ivVehicleImage, ivSelected;
        ConstraintLayout clRoot;

        ViewHolder(View itemView) {
            super(itemView);
            tvVehicleName = (TextView) itemView.findViewById(R.id.tvVehicleName);
            tvBaseFare = (TextView) itemView.findViewById(R.id.tvBaseFare);
            tvFarePerMinute = (TextView) itemView.findViewById(R.id.tvFarePerMinute);
            tvFarePerMile = (TextView) itemView.findViewById(R.id.tvFarePerMile);
            ivVehicleImage = (ImageView) itemView.findViewById(R.id.ivVehicleImage);
            ivSelected = (ImageView) itemView.findViewById(R.id.ivSelected);
            clRoot = (ConstraintLayout) itemView.findViewById(R.id.clRoot);
        }
    }

}

File: Android Studio/JugnooNewUI/jugnooNewUI/src/main/java/product/clicklabs/jugnoo/home/adapters/ScheduleRideVehicleListAdapter.kt
package product.clicklabs.jugnoo.home.adapters

import android.graphics.Typeface.BOLD
import android.support.v7.widget.RecyclerView
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import com.squareup.picasso.Picasso
import kotlinx.android.synthetic.main.list_item_schedule_ride_vehicles.view.*
import product.clicklabs.jugnoo.Data
import product.clicklabs.jugnoo.R
import product.clicklabs.jugnoo.home.HomeActivity
import product.clicklabs.jugnoo.home.models.Region
import product.clicklabs.jugnoo.retrofit.model.Package
import product.clicklabs.jugnoo.retrofit.model.ServiceTypeValue
import product.clicklabs.jugnoo.utils.Fonts
import product.clicklabs.jugnoo.utils.Utils

class ScheduleRideVehicleListAdapter(val activity: HomeActivity, val vehicleList: ArrayList<Region>, val selectedCallback: OnSelectedCallback) : RecyclerView.Adapter<RecyclerView.ViewHolder>() {


    override fun getItemCount(): Int {
        return vehicleList.size
    }


    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        return ViewHolderVehicle(LayoutInflater.from(activity).inflate(R.layout.list_item_schedule_ride_vehicles, parent, false))
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {

        when (holder) {
            is ViewHolderVehicle -> holder.bind(position)
        }


    }

    inner class ViewHolderVehicle(view: View) : RecyclerView.ViewHolder(view) {

        init {
            (view.findViewById(R.id.tvVehicleName) as TextView).setTypeface(Fonts.mavenMedium(activity),BOLD)
            (view.findViewById(R.id.tvBaseFare) as TextView).typeface = Fonts.mavenRegular(activity)
            (view.findViewById(R.id.tvFarePerMinute) as TextView).typeface = Fonts.mavenRegular(activity)
            (view.findViewById(R.id.tvFarePerMile) as TextView).typeface = Fonts.mavenRegular(activity)
        }
        fun bind(position: Int) {
            for (i in vehicleList.indices) {
                if (activity.selectedIdForScheduleRide == vehicleList[position].regionId) {
                    itemView.ivSelected?.visibility = View.VISIBLE
                } else {
                    itemView.ivSelected?.visibility = View.GONE
                }
            }
            itemView.tvVehicleName?.text = vehicleList[position].regionName
            if(vehicleList[position].rideType == ServiceTypeValue.RENTAL.type
                    && vehicleList[position].packages != null
                    && vehicleList[position].packages.size > 0){
                var packageSelected = if (selectedCallback.getPackageSelected() != null) selectedCallback.getPackageSelected() else vehicleList[position].packages[0]
                var match = false
                for(pc in vehicleList[position].packages){
                    if(packageSelected!!.packageId == pc.packageId){
                        match = true
                        break
                    }
                }
                packageSelected = if (match) packageSelected else vehicleList[position].packages[0]

                itemView.tvBaseFare?.text = activity.getString(R.string.package_rate_format, " " + Utils.formatCurrencyValue(Data.autoData.currency,
                        packageSelected!!.fareFixed!!))
                itemView.tvFarePerMinute?.text = activity.getString(R.string.additional_per_min) + ": " + Utils.formatCurrencyValue(Data.autoData.currency, packageSelected.farePerMin!!, false)
                itemView.tvFarePerMile?.text = activity.getString(R.string.additional_per_km_fare, Utils.getDistanceUnit(Data.autoData.distanceUnit)) + ": " + Utils.formatCurrencyValue(Data.autoData.currency, packageSelected.farePerKmAfterThreshold!!, false)
            } else {
                itemView.tvBaseFare?.text = activity.getString(R.string.base_fare_format, " " + Utils.formatCurrencyValue(vehicleList[position].fareStructure.currency,
                        vehicleList[position].fareStructure.getDisplayBaseFare(activity)))
                itemView.tvFarePerMinute?.text = activity.getString(R.string.nl_per_min) + ": " + Utils.formatCurrencyValue(vehicleList[position].fareStructure.currency, vehicleList[position].fareStructure.farePerMin, false)
                itemView.tvFarePerMile?.text = activity.getString(R.string.per_format, Utils.getDistanceUnit(vehicleList[position].fareStructure.distanceUnit)) + ": " + Utils.formatCurrencyValue(vehicleList[position].fareStructure.currency, vehicleList[position].fareStructure.farePerKm, false)
            }

            Picasso.with(activity)
                    .load(vehicleList[position].images.tabHighlighted)
                    .into(itemView.ivVehicleImage)

            itemView.clRoot.tag = position
            itemView.clRoot.setOnClickListener {
                val pos:Int = it.tag as Int
                    vehicleList[pos].run {
                    activity.selectedIdForScheduleRide = regionId!!
                    activity.selectedRideTypeForScheduleRide = rideType!!
                    activity.selectedRegionForScheduleRide = this
                        selectedCallback.onItemSelected(this)
                    notifyDataSetChanged()
                }

            }
        }
    }


    interface OnSelectedCallback{
        fun onItemSelected(selectedRegion: Region)
        fun getPackageSelected(): Package?
    }
}

